projects/js-packages/publicize-components/src/components/add-connection-modal/connect-page/connect-page.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button, useBreakpointMatch } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import { Icon, chevronLeft } from '@wordpress/icons';
+import classNames from 'classnames';
+import { Connection } from '../constants';
+import styles from './style.module.scss';
+
+type ConnectPageProps = {
+	service: Connection;
+	onBackClicked: () => void;
+};
+
+export const ConnectPage: React.FC< ConnectPageProps > = ( { service, onBackClicked } ) => {
+	const [ isSmall ] = useBreakpointMatch( 'sm' );
+
+	return (
+		<>
+			<div
+				className={ classNames( styles[ 'example-wrapper' ], {
+					[ styles.small ]: isSmall,
+				} ) }
+			>
+				{ service.examples.map( ( Example, idx ) => (
+					<div key={ service.name + idx } className={ styles.example }>
+						<Example />
+					</div>
+				) ) }
+			</div>
+			<div className={ styles[ 'actions-wrapper' ] }>
+				<Button variant="secondary" onClick={ onBackClicked }>
+					{ <Icon icon={ chevronLeft } className={ styles[ 'chevron-back' ] } /> }
+				</Button>
+				<form className={ classNames( styles[ 'connect-form' ], { [ styles.small ]: isSmall } ) }>
+					{ 'mastodon' === service.name ? (
```

Comment:
For Mastodon we need to have the username as an input. We can handle that on the Button onClick event